### PR TITLE
Align csub pyaload description in NGSIv2 spec with latest changes in implementation

### PR DESCRIPTION
**Pull request by fgalan**
This PR fixes the NGSIv2 spec .apib in order to align with the latest changes in CB implementation (related with issues #2014, #2042 and #2043).

People that may be insterested in having a look to this:

@iariasleon 
@anabelengp 
@kzangeli 
@crbrox 
@jmcanterafonseca

### CODE REVIEW

File: doc/apiary/v2/fiware-ngsiv2-reference.apib [hunk 1]
```diff
@@ -1343,43 +1343,49 @@ A subscription is represented by a JSON object with the following fields:
...
 + `lastNotification`: Last notification date in ISO8601 format.
 
+An `http` object contains the following subfields:
+
++ `url` : URL pointing to the service which will be invoked when a notification is generated. A NGSIv2
+  compliant server must support `http` URL schema, other schemas could also be supported.
++ `headers` (optional): a key-map of HTTP headers that will be included in the notification messages-
++ `qs` (optional): a key-map of URL query parameters that will be included in the notification message.
++ `method` (optional): the method to use for sending the notification (default is POST)
++ `payload` (optional): the payload to be used in the notification. If omitted, the default payload (see
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: if we have not decided yet, the template mechanism, probably we can leave it out in this PR

File: doc/apiary/v2/fiware-ngsiv2-reference.apib [hunk 2]
```diff
@@ -1343,43 +1343,49 @@ A subscription is represented by a JSON object with the following fields:
...
   `normalized` (the one used if the field is omitted), `keyValues` or `values`. If `attrsFormat` takes any value different
   than those an error will be raised. See detail in "Notification messages" section.
-+ `throttling`: Minimal period of time in seconds which must elapse between two consecutive notifications. It
-  is optional.
 + `timesSent` (not editable, only in GET operations): Number of notifications sent due to this subscription.
 + `lastNotification`: Last notification date in ISO8601 format.
 
+An `http` object contains the following subfields:
+
++ `url` : URL pointing to the service which will be invoked when a notification is generated. A NGSIv2
```
Comment (kzangeli) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: pointing to ...   referencing ?

File: doc/apiary/v2/fiware-ngsiv2-reference.apib [hunk 3]
```diff
@@ -1343,43 +1343,49 @@ A subscription is represented by a JSON object with the following fields:
...
+  compliant server must support `http` URL schema, other schemas could also be supported.
++ `headers` (optional): a key-map of HTTP headers that will be included in the notification messages-
```
Comment (kzangeli) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: minus sign at EOL

Comment (kzangeli) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: I would not use "will be" ... (always makes me wonder '**when** will that be?') 
Instead something more 'strong', like
**is/are to be**, or just **is/are**.

E.g.:

a key-map of HTTP headers that will be included in ...  =>
a key-map of HTTP headers that are included in

File: doc/apiary/v2/fiware-ngsiv2-reference.apib [hunk 4]
```diff
@@ -1343,43 +1343,49 @@ A subscription is represented by a JSON object with the following fields:
...
-+ `attributes`: List of attributes to be included in the notification message. If not specified (or empty), all attributes are included
++ `attrs`: List of attributes to be included in the notification message. If not specified (or empty), all attributes are included
   in the notification. It also defines the order in which attributes will appear in notifications when `attrsFormat` `value` is used
   (see "Notification messages" section).
-+ `callback` : URL pointing to the service which will be invoked when a notification is generated. A NGSIv2
-  compliant server must support `http` URL schema, other schemas (e.g. schemas for web sockets) could also be supported.
-+ `headers` (optional): a key-map of HTTP headers that will be included in the notification messages. Note that this cannot be used
-  to override headers that make the notification inconsistent (e.g. `Content-Type: application/json`).
-+ `query` (optional): a key-map of URL query parameters that will be included in the notification message.
++ `http`: Information about the HTTP transport.
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: it is used to convey parameters for notifications delivered through the HTTP protocol.

Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: Thanks for the suggestion! I was trying to find a more descriptive sentence this morning and, due to lack of ideas, I ended in a very poor one ;)

File: doc/apiary/v2/fiware-ngsiv2-reference.apib (hunk 1 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: The mechanism is more or less decided and it goes in that line. However, if adjustments are needed, we will do them in the .apib in the same PR implementing the functionality, along the code.

Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: NTC

File: doc/apiary/v2/fiware-ngsiv2-reference.apib (hunk 4 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: Fixed in 0f6c2fcf4052402b8851520d06982906314ad72c

File: doc/apiary/v2/fiware-ngsiv2-reference.apib (hunk 3 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: Both fixed in 0f6c2fcf4052402b8851520d06982906314ad72c

File: doc/apiary/v2/fiware-ngsiv2-reference.apib (hunk 2 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: Fixed in 0f6c2fcf4052402b8851520d06982906314ad72c

**Comment by kzangeli** [comment]
LGTM

**Comment by jmcanterafonseca** [comment]
LGTM, although payload subfield is something to be analysed more carefully in further PRs

**Pull request merged**